### PR TITLE
Add reference architecture diagrams for 5k, 10k, 25k user deployments

### PR DESCRIPTION
**Pull request by jwilander**

### CODE REVIEW

File: source/overview/architecture.rst [hunk 1]
```diff
@@ -81,12 +81,60 @@ Moreover, search replicas are also supported to handle search queries.
...
+
+25,000 Users on E20 - AWS 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. image:: ../images/MattermostDeployment25kaws.png
+
+Database with VIPs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following diagram is a suggested configuration for highly-available databases through virutal IPs.
```
Comment (amyblais) [inline source/overview/architecture.rst]:
```suggestion
The following diagram is a suggested configuration for highly-available databases through virtual IPs.
```

File: source/overview/architecture.rst [hunk 2]
```diff
@@ -81,12 +81,60 @@ Moreover, search replicas are also supported to handle search queries.
...
+
+.. image:: ../images/DatabasewithVIPs.png
+
+Load testing
```
Comment (amyblais) [inline source/overview/architecture.rst]:
```suggestion
Load Testing
```

File: source/overview/architecture.rst [hunk 3]
```diff
@@ -81,12 +81,60 @@ Moreover, search replicas are also supported to handle search queries.
...
 
-Reference Architecture
+Reference Architectures
 ----------------------------------
 
+The following diagrams show the suggested architecture configurations for E20 enterprise deployments of Mattermost at different scales. These diagrams are meant as guidelines for typical Mattermost deployments. Hardware and infrastructure requirements can vary significantly based on usage and policies.
+
+There are two versions of each diagram. One is generalized while the other focuses on AWS.
+
+Each generalized diagram represents a full high availability deployment across all critical components. The proxy, database, file storage and Elasticsearch layers can be replaced by cloud services. Push proxy can be replaced by the Mattermost [HPNS](https://docs.mattermost.com/mobile/mobile-hpns.html).
```
Comment (amyblais) [inline source/overview/architecture.rst]: Please check earlier links in the doc on how this link should be formatted.

Comment (jwilander) [reply source/overview/architecture.rst]: Whoops, wrote it in Markdown first and missed this

**Pull request merged**